### ISSUE TITLE
Switch to proxy editing automatically

### ISSUE THREAD

**Issue by raindropsfromsky** [OpenShot/openshot-qt#5464]
When the project becomes ungainly, switching to proxy editing helps the user.
But this decision is left to the user: if he sees lag, he can switch over to proxy editing.

How about letting Openshot a smart decision about this on its own?

Provide this option in the program option. 
Let OS switch over to proxy editing when the project exceeds threshold criteria.
(e.g. xx GB of memory space, when 4k videos are added, performance drops below a threshold FPS, etc.)

**Comment by Colorjet3** [comment]
There are several "Performance Issues" Trello cards to improve the overall performance of OpenShot.  As a matter of fact, I recommend this sometimes to users with large/complex projects and/or with under powered computers.  I let them know to take small chunks of their project, and when done with their edits, to Export and then re-import that .mp4 as input to OpenShot.

I will add this suggestion to one of the Trello cards and the lead dev can decide how to implement this, if he approves it.

I definitely don't like the idea of OpenShot doing it automatically (because you have to know what OpenShot is doing for you automatically).  I'd rather see a some kind of a popup window making a suggestion then pointing you to a very detailed documentation and/or a Tutorial.

**Comment by raindropsfromsky** [comment]
Yes, it is perfectly ok if Openshot senses overload and pops up a message.
Many users feel in control when they get to make every decision.

**Comment by leodf1** [comment]
There is no explicit proxy setting in Openshot to begin with.  You set the Preview resolution in Preferences - Preview

In other Editors the proxy is implemented as a dropdown menu on the Preview screen letting the user select 1/2, 1/4, 1/8, 1/16 playback resolution. I don't know that any Editor does this automatically.

In OpenShot the preview resolution (proxy) can be adjusted by dragging the size of the Preview dock. So a fast and effective method to improve Preview playback is to simply reduce the Preview panel (by enlarging the Timeline panel for example). The downside is that you have a smaller preview window to work with, whereas a proper proxy setting would keep the preview window size.

Optimized Media - This is where Editors create temporary low resolution copies of the media to work on. When exporting, the Editor draws from the original media.  The OpenShot Guide suggests this in a roundabout manual way, by creating low resolution copies of your assets and working  from those. When the project is done, you move the project .osp file to the original media folder and do the export with the original media

**Comment by raindropsfromsky** [comment]
We should be able to keep the preview window as large as we want, but the resolution should be reduced as the user wants.

IMHO, the "proxy" idea is hyped up needlessly! The user does not even need to know that all the editing is being previewed on a lo-res proxy copy and the actual operation will be done on the original files!

As such, the temporary proxy files would be deleted after the job!

It is best to project this feature as "performance accelerator".
Provide a slider (or a drop-down), which varies from x1 to x16.
Let the user think that he is entering turbo mode.

A small caution needs to be shown that a higher acceleration means he cannot see high-resolution videos in the Preview window, but the final quality would remain unaffected. This dialog should have a checkbox: "Do not show this message in future!"

**Comment by Colorjet3** [comment]
There is something that is not sitting well with me regarding this enhancement request.  My brain needs to process this idea/logic, and it will.  I will have to come back and provide my thoughts on this later.
